### PR TITLE
Bug 1866084: AWS EBS parameters for EBS CSI storageclass

### PR DESCRIPTION
**Pull request by cloudbehl**
Made parameters & options available looking at https://github.com/kubernetes-sigs/aws-ebs-csi-driver#createvolume-parameters doc.

588DD614-C5A8-4ED7-9F66-81DC2427793D

Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

### CODE REVIEW

File: frontend/public/components/storage-class-form.tsx [hunk 1]
```diff
@@ -103,6 +103,44 @@ export class StorageClassForm_ extends React.Component<
...
+        type: {
+          name: 'Type',
+          values: { io1: 'io1', gp2: 'gp2', sc1: 'sc1', st1: 'st1', standard: 'standard' },
+          hintText: 'Select AWS Type. Default is gp2',
+        },
+        iopsPerGB: {
+          name: 'IOPS Per GiB',
+          hintText: 'I/O operations per second per GiB',
+          validation: (params) => {
+            if (params.iopsPerGB.value !== '' && !params.iopsPerGB.value.match(/^[1-9]\d*$/)) {
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx:120]: Isn't this the same as `^\d+$`? Are fractional values allowed? (If so, the regex is probably wrong.)

```suggestion
            if (params.iopsPerGB.value && !params.iopsPerGB.value.match(/^\d+$/)) {
```

Comment (spadgett) [reply frontend/public/components/storage-class-form.tsx:120]: `Number.isFinite(Number(params.iopsPerGB.value))` might be better. Although it looks like you've copied what's below.

File: frontend/public/components/storage-class-form.tsx [hunk 2]
```diff
@@ -103,6 +103,44 @@ export class StorageClassForm_ extends React.Component<
...
+              return 'IOPS per GiB must be a number';
+            }
+            return null;
+          },
+          visible: (params) => _.get(params, 'type.value') === 'io1',
+        },
+        fsType: {
+          name: 'Filesystem Type',
+          hintText: 'Select Filesystem to Be Laid Out. Default is ext4',
```
Comment (spadgett) [inline frontend/public/components/storage-class-form.tsx:129]: This is odd case, but I see it's the same as below. Should it be "to be laid out"?

Comment (cloudbehl) [reply frontend/public/components/storage-class-form.tsx:129]: How about going with this?
`Selected File system type that will be formatted during volume creation. Default Type is ext4`

File: frontend/public/components/storage-class-form.tsx (hunk 1 above)
Comment (cloudbehl) [reply frontend/public/components/storage-class-form.tsx:120]: `(/^\d+$/)` && `(/^\d+$/)` is same and fractional values are not allowed looking at code[1].

[1] . https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/44ec7de2c953b52c9fd39e85ca2183aab6465d32/pkg/driver/controller.go#L128

`Number.isFinite(Number(params.iopsPerGB.value))` should I change it at both places?

Comment (spadgett) [reply frontend/public/components/storage-class-form.tsx:120]: `/^\d+$/` is good if we don't allow fractions.

File: frontend/public/components/storage-class-form.tsx (hunk 2 above)
Comment (spadgett) [reply frontend/public/components/storage-class-form.tsx:129]: Maybe "Filesystem type to use during volume creation. Default is ext4." Punctuation on both sentences. Filesystem is one word.

**Pull request merged**